### PR TITLE
Flyswatters

### PR DESCRIPTION
**Pull request by lordpidey**
:cl:
add: Flyswatters are now included in beekeeping crates, they are great at killing insects and flies of all sizes.
/:cl:

Adds flyswatters, useful for keeping bees under control, and also deals massive damage to flypeople.

flyswatter

Mergetoken:  #18511

### CODE REVIEW

**Comment by WJohn** [comment]
Please tell me the swatter pad there has some sort of transparency in the holes so people can click on it easily.

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 1]
```diff
@@ -243,6 +243,12 @@
 		return 1
 	return 0
 
+/mob/living/simple_animal/hostile/poison/bees/attacked_by(obj/item/I, mob/living/user)
+	. = ..()
+	if(istype(I, /obj/item/weapon/melee/flyswatter))
+		user << "<span class='warning'>You easily splat the [src].</span>"
+		new /obj/effect/decal/cleanable/deadcockroach(get_turf(src))
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: bees are now cockroaches.

Why not just called the `death` proc for bees instead?

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 2]
```diff
@@ -243,6 +243,12 @@
...
+		Destroy()
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: NO! Never call `Destroy` directly. *EVER*

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 3]
```diff
@@ -276,6 +282,10 @@
 				name = queen.name
 			else
 				user << "<span class='warning'>You don't have enough units of that chemical to modify the bee's DNA!</span>"
+	if(istype(I,/obj/item/weapon/melee/flyswatter))
+		user << "<span class='warning'>You easily splat the [src].</span>"
+		new /obj/effect/decal/cleanable/deadcockroach(get_turf(src))
+		Destroy()
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: again...calling `Destroy` directly is a no no. use `qdel(src)`

File: code/game/objects/items/weapons/weaponry.dm [hunk 4]
```diff
@@ -313,4 +313,15 @@
...
+/obj/item/weapon/melee/flyswatter
+	name = "Flyswatter"
+	desc = "Useful for killing insects of all sizes."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "flyswatter"
+	item_state = "flyswatter"
+	force = 1
+	throwforce = 1
+	attack_verb = list("swatted", "smacked")
+	w_class = 2
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/weaponry.dm]: Add the mousetrap snap sound as the hitsound for this, just for hilarity, IMO.

**Comment by Cheridan** [comment]
flymen are pure
do not bully the flymen

**Comment by Zilenan91** [comment]
I don't think flyswatters should do 30 damage to flypeople...

**Comment by lordpidey** [comment]
Would 20 damage be better?

**Comment by KorPhaeron** [comment]
>I don't think flyswatters should do 30 damage to flypeople...

That's hilarious please keep it

**Comment by Durkel** [comment]
10/10, anti fly weapon is go.

**Comment by Fox-McCloud** [comment]
Correct me if I'm wrong, but wouldn't unique behaviors be better suited for an `afterattack` rather than putting it into a mob or item's `attack_by`?

**Comment by lordpidey** [comment]
@Fox-McCloud I could easily move it there, but I don't see why it would be more logical there?  Is it a consistency thing?

**Comment by lordpidey** [comment]
Aright, on further review, I will move it to the item's attack_by, since I forgot about another type of insect on the station, butterflies.

**Comment by Armhulen** [comment]
I feel like 30 damage to flies is op
but at the same time if we don't add that the guilt will never die

**Comment by Zilenan91** [comment]
It's in a crate nobody will ever order anyways though so it's probably fine. Maybe in a few years a QM will massacre a bunch of flypeople with a flyswatter and this would be what would make that happen.

**Comment by lordpidey** [comment]
I mean, the beekeeping crate is literally the only way to start beekeeping currently, so it's not gonna be completely impossible.

**Comment by Thunder12345** [comment]
**S W A T   T H E   F L I G G E R S**

**Comment by InsaneHyena** [comment]
Flypeople were a mistake.

**Comment by AdamElTablawy** [comment]
i gotta bee honeyst with you here guys the most important part of this PR is the bee balancing

cause right now bees swarm everywhere and it's quite difficult to stop a massive swarm of initropidil bees

and i mean i know it's probably none of my beesness but i think not merging this would really sting regardless of the feature-without-fix

**Comment by Bawhoppen** [comment]
i literally was like a day away from doing this on my own
 the coincidences are piling up reeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

**Comment by Supermichael777** [comment]
flypeople are older then lizards. a breeding population formed from all the acidents.

**Comment by KazeEspada** [comment]
>unable to swat cockroaches

**Comment by Armhulen** [comment]
pesky bee

**Comment by lordpidey** [comment]
Okay, I finally got around to moving the code to afterattack.

It also supports butterflies/cockroaches now too.

**Comment by kilkun** [comment]
the real question: when are we gonna get an equivalent for ~~argonians~~ lizards?

File: code/game/objects/items/weapons/weaponry.dm [hunk 5]
```diff
@@ -313,4 +313,28 @@
...
+	item_state = "flyswatter"
+	force = 1
+	throwforce = 1
+	attack_verb = list("swatted", "smacked")
+	hitsound = 'sound/effects/snap.ogg'
+	w_class = 2
+
+/obj/item/weapon/melee/flyswatter/afterattack(atom/target, mob/user, proximity_flag)
+	if(proximity_flag)
+		if(istype(target, /mob/living/simple_animal/hostile/poison/bees/) || istype(target, /mob/living/simple_animal/butterfly) || istype(target, /mob/living/simple_animal/cockroach))
```
Comment (AnturK) [inline code/game/objects/items/weapons/weaponry.dm]: Make this typecache so you can expand it easily and varedit goliath swatters

File: code/modules/mob/living/carbon/human/species_types.dm [hunk 6]
```diff
@@ -623,6 +623,11 @@
 						"<span class='userdanger'>You throw up on the floor!</span>")
 	..()
 
+/datum/species/fly/check_weakness(obj/item/weapon, mob/living/attacker)
```
Comment (KorPhaeron) [inline code/modules/mob/living/carbon/human/species_types.dm]: Thought you were moving this to afterattack

Comment (lordpidey) [inline code/modules/mob/living/carbon/human/species_types.dm]: Check_weakness is designed explicitly for things like this.  There is no way to retroactively change the amount of damage done in afterattack.

Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/species_types.dm]: This is true, but you *can* apply damage and check for armor..or you can just make it bypass armor completely and do `take_overall_damage`, `adjustBruteLoss`, or the likes.

**Comment by Metacide** [comment]
So good :+1:

**Comment by tedward1337** [comment]
p r a i s e

**Pull request merged**